lib/nylas/utils/file_utils.rb

```diff
@@ -36,6 +36,13 @@ def self.build_form_request(request_body)
           file = File.open(attachment[:file_path], "rb")
         end
 
+        # Setting original filename and content type if available. See code rest-client#lib/restclient/payload.rb
```

Comment:
Using not fully documented rest-client feature to set the filename in the payload. See https://github.com/rest-client/rest-client/blob/master/lib/restclient/payload.rb#L185